pkg/clients/service.go

```diff
@@ -0,0 +1,72 @@
+package clients
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/monitoring"
+	hClient "github.com/ory/hydra-client-go/v2"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	hydra HydraClientInterface
+
+	tracer  trace.Tracer
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+func (s *Service) parseServiceError(r *http.Response, e error) *hClient.ErrorOAuth2 {
+	se := hClient.NewErrorOAuth2()
+
+	err := unmarshalByteJson(r.Body, se)
+	if err != nil {
+		s.logger.Debugf("Failed to unmarshal JSON: %s", err)
+		return se
+	}
+	se.SetStatusCode(int64(r.StatusCode))
+
+	return se
+}
+
+func (s *Service) GetClient(ctx context.Context, clientID string) (*hClient.OAuth2Client, *hClient.ErrorOAuth2, error) {
+	_, span := s.tracer.Start(ctx, "hydra.OAuth2Api.GetClient")
+	defer span.End()
+
+	c, resp, err := s.hydra.OAuth2Api().
+		GetOAuth2Client(ctx, clientID).
+		Execute()
+
+	if err != nil {
+		return nil, s.parseServiceError(resp, err), nil
+	}
+	return c, nil, nil
+}
+
+func NewService(hydra HydraClientInterface, tracer trace.Tracer, monitor monitoring.MonitorInterface, logger logging.LoggerInterface) *Service {
+	s := new(Service)
+
+	s.hydra = hydra
+
+	s.monitor = monitor
+	s.tracer = tracer
+	s.logger = logger
+
+	return s
+}
+
+func unmarshalByteJson(data io.Reader, v any) error {
+	json_data, err := io.ReadAll(data)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(json_data, v)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
i'd stay away from the `any` generic, better to put `interface{}` or probably do this inside the function inline so that it doesn't feel like magic